source/adapters/level_zero/context.cpp

```diff
@@ -178,111 +183,120 @@ UR_APIEXPORT ur_result_t UR_APICALL urContextSetExtendedDeleter(
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
 
+// Template helper function for creating USM pools for given pool descriptor.
+template <typename P, typename... Args>
+std::pair<ur_result_t, umf::pool_unique_handle_t>
```

Comment:
There's now a helper for this sort of case: https://github.com/oneapi-src/unified-runtime/blob/main/source/ur/ur.hpp#L277

Reply:
I don't know if there's a benefit to using it since I call `umf::poolMakeUnique` that returns a pair anyway.